utils.go

```diff
@@ -238,3 +239,68 @@ func avroEncode[T any](key string, version int, vals []T, out io.Writer) error {
 
 	return enc.Close()
 }
+
+func structTypeToAvroPartitionSchema(st *StructType) (avro.Schema, error) {
+	var aFields []*avro.Field
+
+	for _, field := range st.Fields() {
+		aField, err := nestedFieldToAvroField(field)
+
+		if err != nil {
+			return nil, err
+		}
+
+		aFields = append(aFields, aField)
+	}
+
+	return avro.NewRecordSchema("r102", "", aFields)
```

Comment:
should this be "r102"? Or should this be more dynamic?

Reply:
Not sure. The spec seems not to enforce any specific name here.

Reply:
Here is how Pyiceberg derive the record name: https://github.com/apache/iceberg-python/blob/efc8b5ac0f16717f776e034ecf9a9e9bdabd8424/pyiceberg/utils/schema_conversion.py#L543

Reply:
seems reasonable we should do the same then :)